exercises/robot-simulator/src/main/java/GridPosition.java

```diff
@@ -0,0 +1,16 @@
+public final class GridPosition {
+
+    final int x;
+
+    final int y;
+
+    GridPosition(final int x, final int y) {
+        this.x = x;
+        this.y = y;
+    }
+
+    boolean equals(final GridPosition gridPosition) {
```

Comment:
Here's what a full implementation of `equals` and `hashCode` looks like:

```
@Override
    public boolean equals(final Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        final GridPosition that = (GridPosition) o;

        if (x != that.x) return false;
        return y == that.y;
    }

    @Override
    public int hashCode() {
        int result = x;
        result = 31 * result + y;
        return result;
    }
```

Since this is a class supplied as part of the starter implementation, my thinking was that the above represented a lot of irrelevant detail a user might think they needed to understand in order to proceed with the problem.